src/components/Api.js

```diff
@@ -0,0 +1,97 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`).catch((err) => {
```

Comment:
Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```